File: 101.cpp
struct Solution {
    bool isMirror(TreeNode* l, TreeNode* r){
        if (l == nullptr || r == nullptr){
            return l == nullptr && r == nullptr;
        }
        return l->val == r->val 
                && this->isMirror(l->left, r->right)
                && this->isMirror(l->right, r->left);
    }
    inline bool isSymmetric(TreeNode* root) {
        return this->isMirror(root, root);
    }
};

File: sol/101.py
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        if root is None:
            return True
        l, r = [root], [root]
        while len(l) != 0:
            x = l.pop()
            y = r.pop()
            c = (x is None) + (y is None)
            if c == 1:
                return False
            if c == 2:
                continue
            if x.val != y.val:
                return False
            l.append(x.left)
            l.append(x.right)
            r.append(y.right)
            r.append(y.left)
        return True
